tests/foreman/ui_airgun/test_discoveredhost.py

```diff
@@ -0,0 +1,414 @@
+            module_host_group.name,
+            module_org.name,
+            module_loc.name,
+        )
+        values = session.host.get_details(host_name)
+        assert values['properties']['properties_table']['Status'] == 'OK'
+        assert not session.discoveredhosts.search(
+            'name = {0}'.format(discovered_host_name))
+
+
+@tier2
+def test_positive_update_name(
+        session, module_org, module_loc, module_host_group, discovered_host):
+    """Update the discovered host name and provision it
+
+    :id: 3770b007-5006-4815-ae03-fbd330aad304
+
+    :Setup: Host should already be discovered
+
+    :expectedresults: The hostname should be updated and host should be
+        provisioned
+
+    :CaseLevel: System
+    """
+    discovered_host_name = discovered_host['name']
+    domain_name = module_host_group.domain.read().name
+    new_name = gen_string('alpha').lower()
+    new_host_name = '{0}.{1}'.format(new_name, domain_name)
+    with session:
+        session.organization.select(org_name=module_org.name)
+        session.location.select(loc_name=module_loc.name)
+        discovered_host_values = session.discoveredhosts.wait_for_entity(
+            discovered_host_name)
+        assert discovered_host_values['Name'] == discovered_host_name
+        session.discoveredhosts.provision(
+            discovered_host_name,
+            module_host_group.name,
+            module_org.name,
+            module_loc.name,
+            quick=False,
+            host_values={'host.name': new_name}
+        )
+        assert (session.host.search(new_host_name)[0]['Name']
+                == new_host_name)
+        values = session.host.get_details(new_host_name)
+        assert values['properties']['properties_table']['Status'] == 'OK'
+        assert not session.discoveredhosts.search(
+            'name = {0}'.format(discovered_host_name))
+
+
+@tier2
+@upgrade
+def test_positive_manual_provision_host_with_rule(
+        session, module_org, module_loc, module_host_group):
+    """Create a new discovery rule and manually create a discovered host
+    using that discovery rule.
+
+    Set query as (e.g IP=IP_of_discovered_host)
+
+    :id: 4488ab9a-d462-4a62-a1a1-e5656c8a8b99
+
+    :Setup: Host should already be discovered
+
+    :expectedresults: Host should reboot and provision
+
+    :CaseLevel: Integration
+
+    :CaseImportance: High
+    """
+    host_ip = gen_ipaddr()
+    discovered_host_name = create_discovered_host(ip_address=host_ip)['name']
+    domain = module_host_group.domain.read()
+    with session:
+        session.organization.select(org_name=module_org.name)
+        session.location.select(loc_name=module_loc.name)
+        entities.DiscoveryRule(
+            max_count=1,
+            hostgroup=module_host_group,
+            search_='ip = {0}'.format(host_ip),
+            location=[module_loc],
+            organization=[module_org],
+        ).create()
+        session.discoveredhosts.apply_action(
```

Comment:
why title is `manual_provision`, but we do `Auto Provision? It is just not logical. I feel you are doing right things, so maybe we just need to rephrase title or?